### PR TITLE
add handleWith test case

### PR DESCRIPTION
**Pull request by azole**

### CODE REVIEW

File: src/ActionKit/ActionRunner.php [hunk 1]
```diff
@@ -107,7 +111,50 @@ public function run($actionName, $arguments = array() )
...
+            throw new InvalidActionNameException("");
+        }
+        if ( ! Utils::validateActionName( $request->getActionName() ) ) {
+            throw new InvalidActionNameException( "Invalid action name: " . $request->getActionName() . ".");
+        }
+
+        return $this->run($request->getActionName(), $request->getArguments());
+    }
+
+    public function handleWith($stream, $arguments = array())
```
Comment (c9s) [inline src/ActionKit/ActionRunner.php]: Add type hinting: `array $arguments = array()`

File: src/ActionKit/ActionRunner.php [hunk 2]
```diff
@@ -107,7 +111,50 @@ public function run($actionName, $arguments = array() )
             return $this->results[ $actionName ] = $action->getResult();
         }
 
-        throw new Exception( "Can not create action class $class" );
+        throw new UnableToCreateActionException( "Can not create action class $class" );
+    }
+
+    public function runWith($request)
```
Comment (c9s) [inline src/ActionKit/ActionRunner.php]: type hinting: `(ActionRequest $request)`

File: src/ActionKit/ActionRunner.php [hunk 3]
```diff
@@ -107,7 +111,50 @@ public function run($actionName, $arguments = array() )
...
-        throw new Exception( "Can not create action class $class" );
+        throw new UnableToCreateActionException( "Can not create action class $class" );
+    }
+
+    public function runWith($request) 
+    {
+        if (!$request->getActionName()) {
+            throw new InvalidActionNameException("");
+        }
+        if ( ! Utils::validateActionName( $request->getActionName() ) ) {
```
Comment (c9s) [inline src/ActionKit/ActionRunner.php]: Move `validateActionName` into ActionRequest class?

File: src/ActionKit/ActionRunner.php [hunk 4]
```diff
@@ -95,7 +99,7 @@ public function __construct($options = array()) {
     public function run($actionName, $arguments = array() )
```
Comment (c9s) [inline src/ActionKit/ActionRunner.php]: Type hinting: `array $arguments = array()`

File: src/ActionKit/ActionRequest.php [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        }
+
+        unset($this->arguments['__action']);
+        unset($this->arguments['action']);
+
+        // handle actionName
+        $actionKey = isset($request['__action']) ? '__action' : 'action';
+        if (isset($request[$actionKey])) {
+            $this->actionName = $request[$actionKey];
+        }
```
Comment (c9s) [inline src/ActionKit/ActionRequest.php]: else {  .. throw MissingActionFieldException;  }

File: src/ActionKit/ActionRunner.php [hunk 6]
```diff
@@ -107,7 +111,50 @@ public function run($actionName, $arguments = array() )
...
+            @header('HTTP/1.0 403');
+            if ( $request->isAjax() ) {
+                fwrite($stream, json_encode(array(
+                        'error' => 1,
+                        'message' => $e->getMessage(),
+                        'line' => $e->getLine(),
+                        'file' => $e->getFile(),
+                )));
+            } 
+            throw $e;
```
Comment (c9s) [inline src/ActionKit/ActionRunner.php]: } else { throw $e }

File: src/ActionKit/ActionRunner.php (hunk 3 above)
Comment (azole) [inline src/ActionKit/ActionRunner.php]: throw exception from ActionRequest constructor? 
if throw exception from ActionRequest constructor, $request will be undefined variable in L148 (if ( $request->isAjax() ) {...} ).

**Pull request merged**